src/visitable.cpp

```diff
@@ -103,13 +112,11 @@ static int has_quality_from_vpart( const vehicle& veh, int part, const quality_i
 
             // does the part provide this quality?
             if( iter != tq.end() && iter->second >= level ) {
-                if( ++qty >= limit ) {
-                    break;
```

Comment:
The break is a fairly pointless optimization with questionable payoff, it does terminate early, but at the cost of additional branching every time through the loop, so I'm not confident it's an improvement, so I went with the simpler approach without an early exit.